tests/codeception/functional/MegaMenuCest.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+use Drupal\Core\Url;
+use Faker\Factory;
+
+/**
+ * Class MegaMenuCest.
+ *
+ * @group install
+ */
+class MegaMenuCest {
+
+  /**
+   * Faker service.
+   *
+   * @var \Faker\Generator
+   */
+  protected $faker;
+
+  /**
+   * Test constructor.
+   */
+  public function __construct() {
+    $this->faker = Factory::create();
+  }
+
+  public function _before(FunctionalTester $I) {
+    $I->resizeWindow(2000, 1400);
+    $I->logInWithRole('administrator');
+    $I->amOnPage('/admin/config/site-options');
+    $I->see('Enable New Mega Menu');
+
+    $this->megaMenuEnabled = (bool) $I->grabAttributeFrom('[name="field_en_mega_menu[value]"]', 'checked');
+    if (!$this->megaMenuEnabled) {
+      $I->checkOption('Enable New Mega Menu');
+      $I->click('Save');
+      drupal_flush_all_caches();
+    }
+  }
+
+  /**
+   * Every main menu item should not error.
+   */
+  public function testMegaMenu(FunctionalTester $I) {
+
+    $topLevelTitle = $this->faker->words(3, TRUE);
+    $secondLevelTitle = $this->faker->words(3, TRUE);
+
+    $I->logInWithRole('administrator');
+    $top_level = $I->createEntity([
+      'title' => $topLevelTitle,
+      'type' => 'hs_basic_page',
+    ]);
+
+    $I->amOnPage($top_level->toUrl('edit-form')->toString());
+    $I->click('.menu-link-form summary');
+    $I->checkOption('Provide a menu link');
+    $I->fillField('Menu link title', $topLevelTitle);
+    $I->scrollTo(['css' => '.form-submit']);
+    $I->click('Save');
+
+    $second_level = $I->createEntity([
+      'title' => $secondLevelTitle,
+      'type' => 'hs_basic_page',
+    ]);
+    $I->amOnPage($second_level->toUrl('edit-form')->toString());
+    $I->click('.menu-link-form summary');
+    $I->checkOption('Provide a menu link');
+    $I->fillField('Menu link title', $secondLevelTitle);
+    $I->scrollTo(['css' => '.form-item-menu-menu-parent'], 0, -100);
+    $I->wait(2);
+    $I->selectOption('Parent link', "-- {$topLevelTitle}");
+    $I->waitForText('Show row weights');
+    $I->click('Show row weights');
+    $I->wait(2);
```

Comment:
Replace `wait`

Reply:
I will try to remove this one but this one has caused many of my issues in the past for making these tests pass. Locally, if I remove this wait, it works great. When it runs on CI, it fails to correctly add this as a child item unless there is a little bit of a wait there. But, I will try and remove it and test it out. There is no text to wait for, somehow it simply needs to wait a little bit on CI. Will let you know.

Reply:
It worked, so this is now resolved.